assembl/views/api2/discussion.py

```diff
@@ -1706,16 +1905,149 @@ def phase2_csv_export(request):
+    ideas = get_bright_mirror_ideas(discussion)
+    row_list = list()
+    for idea in ideas:
+        row = {}
+        row[IDEA_ID] = idea.id
+        row[IDEA_PARENT] = get_idea_parents(idea, user_prefs)
+        row[IDEA_NAME] = get_entries_locale_original(idea.title).get('entry')
+        posts = get_published_posts(idea)
+        for post in posts:
+            if has_lang:
+                post.maybe_translate(target_locales=[language])
+
+            subject = get_entries_locale_original(post.subject)
+            body = get_entries_locale_original(post.body)
+            row[POST_SUBJECT] = subject.get('entry')
+            row[POST_BODY] = sanitize_text(body.get('entry'))
+            row[WORD_COUNT] = str(len(row[POST_BODY].split()))
+            if not has_anon:
+                row[POST_CREATOR_NAME] = post.creator.real_name()
+                row[POST_CREATOR_USERNAME] = post.creator.username_p or ""
+            else:
+                row[POST_CREATOR_NAME] = post.creator.anonymous_name()
+                row[POST_CREATOR_USERNAME] = post.creator.anonymous_username() or ""
+            row[POST_CREATOR_EMAIL] = post.creator.get_preferred_email(anonymous=has_anon)
+            row[POST_CREATION_DATE] = format_date(post.creation_date)
+            row[POST_LIKE_COUNT] = post.like_count
+            row[POST_DISAGREE_COUNT] = post.disagree_count
+            row[POST_DONT_UNDERSTAND_COUNT] = post.dont_understand_count
+            row[POST_MORE_INFO_PLEASE_COUNT] = post.more_info_count
+            if extra_columns_info:
+                if post.creator_id not in column_info_per_user:
+                    column_info_per_user[post.creator_id] = get_social_columns_from_user(
+                        post.creator, extra_columns_info, provider_id)
+                extra_info = column_info_per_user[post.creator_id]
+                for num, (name, path) in enumerate(extra_columns_info):
+                    row[name] = extra_info[num]
+            if not post.sentiments:
+                row[SENTIMENT_ACTOR_NAME] = u''
+                row[SENTIMENT_ACTOR_EMAIL] = u''
+                row[SENTIMENT_CREATION_DATE] = u''
+                writer.writerow(convert_to_utf8(row))
+            if not post.shares:
+                row[NUMBER_OF_SHARES] = u''
+            else:
+                row[NUMBER_OF_SHARES] = str(post.get_number_of_shares)
+            for sentiment in post.sentiments:
+                if not has_anon:
+                    row[SENTIMENT_ACTOR_NAME] = sentiment.actor.real_name()
+                else:
+                    row[SENTIMENT_ACTOR_NAME] = sentiment.actor.anonymous_name()
+                row[SENTIMENT_ACTOR_EMAIL] = sentiment.actor.get_preferred_email(anonymous=has_anon)
+                row[SENTIMENT_CREATION_DATE] = format_date(
+                    sentiment.creation_date)
+                row_list.append(convert_to_utf8(row))
+    return fieldnames, row_list
```

Comment:
Is there really lots of difference between thread, survey, brightMirror exports? Can't we have only one code, eventually adding some condition for some fields?

Reply:
@vincentfretin  I thought about it but finally I found the code will be more modular and easily maintainable than if we put everything in one function.

Reply:
Personally, I don't find it maintainable. If you want to add a field to the export, you will need to modify the code at four different places.